File: c/algorithms/sort/selection/iterative.c
#include "../sortlib.h"

int array[10] = { 188, 9, 7, 1, 5, 7777, 4, 3, 1, -99 };
unsigned int short len = sizeof(array) / sizeof(int);

void swap(unsigned int i, unsigned int j) {
    unsigned int short tmp = array[i];
    array[i] = array[j];
    array[j] = tmp;
}

void selectionSort() {
    unsigned int short i, j;

    for (i = 0; i < len; i++) {
        for (j = i + 1; j < len; j++) {
            if (array[j] < array[i]) {
                swap(i, j);
            }
        }
    }
}

void main(void) {
    printArray("Before sort:", array, len);
    selectionSort();
    printArray("After sort:", array, len);
}

File: python/algorithms/sort/selection/iterative.py
import sys

def swap(toSort, i, j):
    tmp = toSort[i]
    toSort[i] = toSort[j]
    toSort[j] = tmp

def selectionSort(toSort):
    length = len(toSort)

    for i in range(0, length):
        for j in range(i + 1, length):
            if toSort[j] < toSort[i]:
                swap(toSort, i, j)

    return toSort

def main(argv):
    toSort = [188, 9, 7, 1, 5, 7777, 4, 3, 1, -99]
    print('initial sort order', toSort)
    print('            sorted', selectionSort(toSort))

if __name__ == '__main__':
    main(sys.argv[1:])

